File: jdbc-mysql/src/main/java/com/example/jdbc/mysql/Author.java
package com.example.jdbc.mysql;

import java.sql.ResultSet;
import java.sql.SQLException;

import org.springframework.jdbc.core.RowMapper;

record Author(long id, String name) {
	enum Mapper implements RowMapper<Author> {

		INSTANCE;

		@Override
		public Author mapRow(ResultSet rs, int rowNum) throws SQLException {
			return new Author(rs.getInt("id"), rs.getString("name"));
		}

	}
}

File: data/data-jdbc-h2-kotlin/src/main/kotlin/com/example/data/jdbc/h2/kotlin/model/Author.kt
package com.example.data.jdbc.h2.kotlin.model

import org.springframework.data.annotation.Id

data class Author(
	@Id val id: Long?,
	val name: String,
	val books: Set<Book>
) {
	override fun toString(): String {
		return "Author(name='$name')"
	}
}
